### ISSUE TITLE
On vscode 1.25 ElixirLs seems to crash

### ISSUE THREAD

**Issue by tcoopman** [JakeBecker/vscode-elixir-ls#81]
When I start VSCode everything is ok, but it starts failing fast.

I'm using elixir 1.6.6 with erlang 21.0.2. A restart fixes it shortly.
I tried removing the .elixir_ls folder but that didn't help.

```
 WARN Aborted format on save after 750ms
2console.ts:136 [Extension Host] Notification handler 'window/logMessage' failed with message: Channel has been closed
t.log @ console.ts:136
t._logExtensionHostMessage @ extensionHost.ts:393
(anonymous) @ extensionHost.ts:210
emitTwo @ events.js:106
emit @ events.js:194
process.nextTick @ internal/child_process.js:766
_combinedTickCallback @ internal/process/next_tick.js:73
_tickCallback @ internal/process/next_tick.js:104
log.ts:171   ERR Channel has been closed: Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.error (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1592:28)
	at LanguageClient.logFailedRequest (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:2050:14)
	at client.sendRequest.then (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:892:24)
	at <anonymous>
9console.ts:136 [Extension Host] Notification handler 'window/logMessage' failed with message: Channel has been closed
t.log @ console.ts:136
t._logExtensionHostMessage @ extensionHost.ts:393
(anonymous) @ extensionHost.ts:210
emitTwo @ events.js:106
emit @ events.js:194
process.nextTick @ internal/child_process.js:766
_combinedTickCallback @ internal/process/next_tick.js:73
_tickCallback @ internal/process/next_tick.js:104
log.ts:165  WARN Aborted format on save after 750ms
console.ts:136 [Extension Host] Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.error (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1592:28)
	at LanguageClient.handleConnectionError (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1884:18)
	at errorHandler (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1849:18)
	at connection.onError (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:35:36)
	at CallbackList.invoke (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:71:39)
	at Emitter.fire (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:135:36)
	at writeErrorHandler (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/main.js:230:22)
	at CallbackList.invoke (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:71:39)
	at Emitter.fire (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:135:36)
	at StreamMessageWriter.AbstractMessageWriter.fireError (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/messageWriter.js:47:27)
	at Socket.<anonymous> (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/messageWriter.js:78:68)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at emitErrorNT (net.js:1283:8)
	at _combinedTickCallback (internal/process/next_tick.js:80:11)
	at process._tickCallback (internal/process/next_tick.js:104:9)
t.log @ console.ts:136
t._logExtensionHostMessage @ extensionHost.ts:393
(anonymous) @ extensionHost.ts:210
emitTwo @ events.js:106
emit @ events.js:194
process.nextTick @ internal/child_process.js:766
_combinedTickCallback @ internal/process/next_tick.js:73
_tickCallback @ internal/process/next_tick.js:104
console.ts:136 [Extension Host] Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.info (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1574:28)
	at LanguageClient.handleConnectionClosed (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1875:18)
	at LanguageClient.handleConnectionClosed (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/main.js:106:15)
	at closeHandler (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1852:18)
	at CallbackList.invoke (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:71:39)
	at Emitter.fire (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:135:36)
	at closeHandler (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/main.js:221:26)
	at CallbackList.invoke (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:71:39)
	at Emitter.fire (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/events.js:135:36)
	at StreamMessageWriter.AbstractMessageWriter.fireClose (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/messageWriter.js:57:27)
	at Socket.<anonymous> (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-jsonrpc/lib/messageWriter.js:79:63)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at Pipe._handle.close [as _onclose] (net.js:510:12)
t.log @ console.ts:136
t._logExtensionHostMessage @ extensionHost.ts:393
(anonymous) @ extensionHost.ts:210
emitTwo @ events.js:106
emit @ events.js:194
process.nextTick @ internal/child_process.js:766
_combinedTickCallback @ internal/process/next_tick.js:73
_tickCallback @ internal/process/next_tick.js:104
log.ts:171   ERR Channel has been closed: Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.error (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1592:28)
	at LanguageClient.logFailedRequest (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:2050:14)
	at client.sendRequest.then (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1050:24)
	at <anonymous>
log.ts:171   ERR Channel has been closed: Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.error (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1592:28)
	at LanguageClient.logFailedRequest (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:2050:14)
	at client.sendRequest.then (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:892:24)
	at <anonymous>
log.ts:171   ERR Channel has been closed: Error: Channel has been closed
	at e.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:372)
	at e.appendLine (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:583:6)
	at LanguageClient.error (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1592:28)
	at LanguageClient.logFailedRequest (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:2050:14)
	at client.sendRequest.then (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1050:24)
	at <anonymous>
log.ts:171   ERR Language client is not ready yet: Error: Language client is not ready yet
	at LanguageClient.sendRequest (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1473:19)
	at provideDocumentFormattingEdits (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1049:27)
	at Object.provideDocumentFormattingEdits (/home/thomas/.vscode/extensions/JakeBecker.elixir-ls-0.2.19/node_modules/vscode-languageclient/lib/client.js:1059:23)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:722:532
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:109:515
	at new n.Class.derive._oncancel (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:84:560)
	at Object.t.asWinJsPromise (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:109:478)
	at e.provideDocumentFormattingEdits (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:722:486)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:735:944
	at e._withAdapter (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:731:534)
	at e.$provideDocumentFormattingEdits (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:735:906)
	at e._doInvokeHandler (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:658:87)
	at e._invokeHandler (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:657:779)
	at e._receiveRequest (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:656:851)
	at e._receiveOneMessage (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:656:620)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:655:544
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:658:673
	at /opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:96:599
	at e.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:97:917)
	at u (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:159:721)
	at Socket.<anonymous> (/opt/visual-studio-code/resources/app/out/vs/workbench/node/extensionHostProcess.js:159:898)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at readableAddChunk (_stream_readable.js:178:18)
	at Socket.Readable.push (_stream_readable.js:136:10)
	at Pipe.onread (net.js:560:20)
```

**Comment by JakeBecker** [comment]
I just published v0.2.19 that should improve error reporting. Please update and try it out, and if it doesn't work, check the "Output" pane and select "ElixirLS" and post anything there that looks relevant.

**Comment by tcoopman** [comment]
Could not reproduce anymore today. I'll close this for now. Thanks
